File: services/17-spring-boot/src/test/java/com/example/controller/HelloControllerTest.java
package com.example.controller;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.test.web.servlet.MockMvc;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;

@WebMvcTest(HelloController.class)
class HelloControllerTest {

    @Autowired
    private MockMvc mvc;

    @Test void testHello() throws Exception {
        mvc.perform(get("/")).andExpect(status().isOk()).andExpect(jsonPath("$.message").exists());
    }

    @Test void testHealth() throws Exception {
        mvc.perform(get("/health")).andExpect(status().isOk()).andExpect(jsonPath("$.status").value("ok"));
    }

    @Test void testLiveness() throws Exception {
        mvc.perform(get("/health/live")).andExpect(status().isOk());
    }

    @Test void testReadiness() throws Exception {
        mvc.perform(get("/health/ready")).andExpect(status().isOk());
    }
}

File: services/18-spring-boot-kotlin/src/test/kotlin/com/example/HelloControllerTest.kt
package com.example

import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.*

@WebMvcTest(HelloController::class)
class HelloControllerTest {
    @Autowired lateinit var mvc: MockMvc

    @Test fun testHello() { mvc.perform(get("/")).andExpect(status().isOk).andExpect(jsonPath("$.message").exists()) }
    @Test fun testHealth() { mvc.perform(get("/health")).andExpect(status().isOk).andExpect(jsonPath("$.status").value("ok")) }
    @Test fun testLiveness() { mvc.perform(get("/health/live")).andExpect(status().isOk) }
    @Test fun testReadiness() { mvc.perform(get("/health/ready")).andExpect(status().isOk) }
}
